### DISCUSSION TITLE
Pre-trained model for inference

### DISCUSSION THREAD

**Discussion by rohit7044** [torchgeo/torchgeo#1607] [General]
First of all thank you for this wonderful work. This is a beautiful repo and I am pretty sure many people gets a lot of benefits. However due to lack of proper documentation and me being new to this field, I am having a hard time understanding tutorial.
Basically I have an basic plain RGB image and I don't have any satellite data. I want to apply various segmentation methods/models to remove vegetation, roads, buildings,trees,terrain,lanes, Can somebody give me any headsup maybe links or a tutorial on how to do that?
Thank You

**Comment by roybenhayun** [comment]
Hi Rohit, as a new comer to TrochGeo myself I can share the learning path I went through. 
1. do the basic getting started tutorials. currently there are [Getting Started](https://torchgeo.readthedocs.io/en/stable/tutorials/getting_started.html#), [Custom Raster Datasets](https://torchgeo.readthedocs.io/en/stable/tutorials/custom_raster_dataset.html),     [Transforms](https://torchgeo.readthedocs.io/en/stable/tutorials/transforms.html), [Indices](https://torchgeo.readthedocs.io/en/stable/tutorials/indices.html), [Lightning Trainers](https://torchgeo.readthedocs.io/en/stable/tutorials/trainers.html),     [Pretrained Weights](https://torchgeo.readthedocs.io/en/stable/tutorials/pretrained_weights.html). These are all basic and must-know. don't take long and pretty easy to follow.
2. watch the [TorchGeo intro](https://www.youtube.com/watch?v=ET8Hb_HqNJQ) by Caleb 
3. watch [IADF School 2022: Change Detection (TorchGeo)](https://www.youtube.com/watch?v=R_FhY8aq708) also by Caleb Robinson
4. read the [original paper](https://arxiv.org/pdf/2111.08872.pdf). it's a must read IMHO.

there is probably more but these were the first steps I took before embarking on a real project. to start your own project, find an example that does something similar to what you need and work it from there. 

I work on semantic segmentation and started from https://github.com/microsoft/satellite-imagery-labeling-tool/blob/main/example/training-example.ipynb

my 2c. HTH.

**Reply by rohit7044** [reply]
Thank you for the reply. However I am not looking for training a model or some sort. I just want to perform inference on  an RGB image and segment various classes. That's the only scope between me and torchgeo right now. Do you have any leads on that?

**Reply by roybenhayun** [reply]
so you want to run prediction, but without training, right?
hmm. haven't seen this yet and would be also interested to see an answer on this.

**Reply by adamjstewart** [reply]
I would add to this list that TorchGeo uses the same API as torchvision/etc., so if you have never used torchvision before, it helps to go through some of their tutorials. We're planning on adding a lot more tutorials in the near future, we want to start having official tutorials at popular ML/RS conferences.

**Comment by rohit7044** [comment]
Yes,
We can preprocess the image according to the model input and run inference
since Torchgeo already has pre-trained weights. I am looking for some
headsup or any information if it's even possible on this library.

Regards,
RD

**Reply by adamjstewart** [reply]
It's easy to do. You could:

1. Instantiate a model
2. Load the pre-trained weights
3. Loop over all images
4. Make predictions for all images

However, none of our pre-trained weights are pre-trained for semantic segmentation, and certainly not for the exact set of classes you need to use. You'll still need a labeled dataset to fine-tune the pre-trained weights. Pre-trained weights are designed to make this quicker and easier, not to skip it altogether.

**Reply by rohit7044** [reply]
Aah! Interesting
Thanks for this information
